libraries/Microsoft.Bot.Builder/Integration/BotFrameworkOptions.cs

```diff
@@ -43,6 +44,17 @@ public BotFrameworkOptions()
         /// <seealso cref="BotAdapter.Use(IMiddleware)"/>
         public IList<IMiddleware> Middleware { get; } = new List<IMiddleware>();
 
+        /// <summary>
+        /// Gets a list of the <see cref="IMiddleware"/> to use on each incoming activity.
+        /// </summary>
+        /// <value>The middleware list.</value>
+        /// <seealso cref="BotAdapter.Use(IMiddleware)"/>
+        public IList<BotState> State { get; } = new List<BotState>();
+
+        public UserState UserState => this.State.OfType<UserState>().First();
```

Comment:
I don't think we need these anymore. With the accessors now passed into the Bot, the average Bot won't need them.